bin/run-circleci-tests

```diff
@@ -4,6 +4,18 @@ set -e
 
 make install_system_dependencies
 
+# Ensure Redis is up and running before starting to run any tests:
+# In the past, individual tests from our unit test suite
+# would sometimes fail with a "ConnectionRefusedError"
+# caused by Redis not starting for the container running the tests.
+# The solution that we use below comes from the following support thread:
+# https://discuss.circleci.com/t/redis-server-problems-unable-to-connect-localhost-6379/4040
+until bash -c "sudo service redis-server status"; do
```

Comment:
@jbzdak Thanks for the review! I'm not sure I understand what you mean -- should the script stop trying to start Redis after 15 minutes have passed? That is, should we just do something like this:

```sh
for i in {1..900}; do
    >&2 echo "Redis not running, trying to kick start it"
    bash -c "sudo service redis-server start" && break || sleep 1;
done
```

Or did you have something else in mind?